ui/src/app/utils/access-utils.tsx

```diff
@@ -222,7 +204,6 @@ export const getAccessModuleConfig = (
       () => ({
         ...apiConfig,
         isEnabledInEnvironment: true,
-        DARTitleComponent: () => <div>Turn on Google 2-Step Verification</div>,
```

Comment:
Why this PR works:

Every re-render of `module.tsx` called `getAccessModuleConfig` which returned a **new instance** of a function that returns a `div`, i.e. a React component. When you used `<DARTitleComponent>` in `module.tsx`, React recognized that the function instance changed and considered it a **different Component** and therefore re-mounted it.

React recognizes uses of `DARTitle` as the same Component and therefore does not re-mount `DARTitle` uses on re-renders.

As a rule of thumb, I recommend only returning JSX in top-level Components.

Reply:
Were individual modules re-rendering or is it that each individual module (all eight of them) was calling getAccessModuleConfig? Or do you suspect that it is a combination of both? 

In either case, I think that this is a great refactor. Good work!

Reply:
Both? `module.tsx` instances (one per module) are re-rendered many times. It calls `getAccessModuleConfig` every render.